src/priority_enumerator.jl

```diff
@@ -45,22 +76,24 @@ function _find_next_complete_tree(grammar::ContextFreeGrammar, max_depth::Int, p
             # We add the potential expanded trees to the pq and move on to 
             # the next tree in the queue.
             for expanded_tree ∈ expanded_trees
-                enqueue!(pq, expanded_tree, size + 1)
+                # Pass the local scope to the function for calculating the priority 
+                enqueue!(pq, expanded_tree, priority_function(Base.@locals))
```

Comment:
@sebdumancic @THinnerichs I'm not really sure about what the best way is of passing information to the function for calculating the priority.
- We can just give the function the available values via its arguments. The downside to this is that when we want to pass additional information for a specific priority function, we need to add an argument and all existing priority functions do not work anymore.
- A more flexible option would be to manually create a dictionary. This provides more control over what gets passed to the priority function.
- Currently, the priority function gets a dictionary of the entire local scope (`Base.@locals`). The benefit is that we do not have to do any manual bookkeeping, and any additional variables we create in this function can automatically be used in the priority function. The drawback is that renaming a variable could break existing priority functions.
- We could also create a special composite type that contains the data that is passed to the priority function/

Reply:
Hmmm, I'm wondering if this is the question for context-sensitive grammar rather than context-free one?  Context free grammar should ignore everything except the current (expanded) expressions and possible next expansions, but nothing more. Perhaps that is all we have to pass here? Then do this thing with locals in context-sensitive grammar which should include any available information.

Reply:
Good idea, I reverted back to only providing the tree and the heuristic value of the parent to the heuristic function. The latter speeds up heuristic calculations since we do not have to calculate the size and depth of trees, so I decided to keep that.